iota-client/src/client.rs

```diff
@@ -396,10 +393,39 @@ impl Client {
     /// Gets the miner to use based on the PoW setting
     pub fn get_pow_provider(&self) -> ClientMiner {
         ClientMinerBuilder::new()
-            .with_local_pow(self.network_info.read().unwrap().local_pow)
+            .with_local_pow(self.get_local_pow())
             .finish()
     }
 
+    /// Gets the network related information and if it's the default one, sync it first.
+    pub async fn get_synced_network_info(&self) -> Result<NetworkInfo> {
+        if self.get_network_info().network_id.is_none() {
+            let info = self.get_info().await?;
+            let network_id = hash_network(&info.network_id);
+            let mut client_network_info = self.network_info.write().unwrap();
+            client_network_info.network_id = Some(network_id);
+            client_network_info.min_pow_score = info.min_pow_score;
+            client_network_info.bech32_hrp = info.bech32_hrp.clone()
+        }
+
+        Ok(self.get_network_info())
+    }
+
+    /// returns the tips interval
+    pub fn get_tips_interval(&self) -> u64 {
+        self.network_info.read().unwrap().tips_interval
+    }
+
+    /// returns the local pow
+       pub fn get_local_pow(&self) -> bool {
+        self.network_info.read().unwrap().local_pow
+    }
+
+    /// returns the min pow score
+    pub fn get_min_pow_score(&self) -> f64 {
+        self.network_info.read().unwrap().min_pow_score
+    }
+
     /// Gets the network related information such as network_id and min_pow_score
     pub fn get_network_info(&self) -> NetworkInfo {
```

Comment:
Maybe we should remove that? Because for the values that aren't dependent on the node you added new values and here someone should rather use the synced network_info